### PR TITLE
Introduce `neo4j.hasReachableServer` API

### PR DESCRIPTION
**Pull request by bigmontz**
This is an experimental API responsible for verifying if the given url is running a Neo4j Server.

Co-Authored-by: Oskar Damkjaer <oskar.damkjaer@neo4j.com>

### CODE REVIEW

File: packages/neo4j-driver-lite/src/index.ts [hunk 1]
```diff
@@ -328,6 +328,26 @@ function driver (
   }
 }
 
+/**
+ * Method which veirfies if the driver is server is reachable.
```
Comment (OskarDamkjaer) [inline packages/neo4j-driver-lite/src/index.ts]:
```suggestion
 * Verifies if the driver can reach a server at the given url.
```

File: packages/neo4j-driver-lite/src/index.ts [hunk 2]
```diff
@@ -328,6 +328,26 @@ function driver (
...
+ *
+ * @experimental
+ * @since 5.0.0
+ * @param {string} url The URL for the Neo4j database, for instance "neo4j://localhost" and/or "bolt://localhost"
+ * @param {Pick<Config, 'logging'>} config Configuration object. See the {@link driver}
+ * @returns {true} When the server is reachable
+ * @throws {Error} When the server is not reacheable or the url is invalid
```
Comment (OskarDamkjaer) [inline packages/neo4j-driver-lite/src/index.ts]:
```suggestion
 * @throws {Error} When the server is not reachable or the url is invalid
```

File: packages/neo4j-driver/src/index.js [hunk 3]
```diff
@@ -305,6 +305,26 @@ function driver (url, authToken, config = {}) {
   }
 }
 
+/**
+ * Method which veirfies if the driver is server is reachable.
```
Comment (OskarDamkjaer) [inline packages/neo4j-driver/src/index.js]: If you agree with the suggested doc change earlier, it'd need to be changed here as well

File: packages/neo4j-driver/src/index.js [hunk 4]
```diff
@@ -305,6 +305,26 @@ function driver (url, authToken, config = {}) {
...
+ *
+ * @experimental
+ * @since 5.0.0
+ * @param {string} url The URL for the Neo4j database, for instance "neo4j://localhost" and/or "bolt://localhost"
+ * @param {object} config Configuration object. See the {@link driver}
+ * @returns {true} When the server is reachable
+ * @throws {Error} When the server is not reacheable or the url is invalid
```
Comment (OskarDamkjaer) [inline packages/neo4j-driver/src/index.js]:
```suggestion
 * @throws {Error} When the server is not reachable or the url is invalid
```

**Review by OskarDamkjaer — APPROVED** [review]
Looking good! Excited to use this in the next release 🎉 Left some suggestions on the doc comment

File: packages/neo4j-driver/test/driver.test.js [hunk 5]
```diff
@@ -588,6 +588,20 @@ describe('#integration driver', () => {
...
+    expect(result).toBe(true)
+  })
+
+  it('hasReachableServer failure', async () => {
+    try {
+      await neo4j.hasReachableServer(`${sharedNeo4j.scheme}://${sharedNeo4j.hostname}:9999`)
+      expect(true).toBe('false')
+    } catch (error) {
+      expect(error).toBeInstanceOf(Error)
+    }
```
Comment (robsdedude) [inline packages/neo4j-driver/test/driver.test.js]: Doesn't that catch the assertion error should the assertion fail?

File: packages/neo4j-driver/test/driver.test.js [hunk 6]
```diff
@@ -588,6 +588,20 @@ describe('#integration driver', () => {
...
 
+  it('hasReachableServer success', async () => {
+    const result = await neo4j.hasReachableServer(`${sharedNeo4j.scheme}://${sharedNeo4j.hostname}`)
+    expect(result).toBe(true)
+  })
+
+  it('hasReachableServer failure', async () => {
+    try {
+      await neo4j.hasReachableServer(`${sharedNeo4j.scheme}://${sharedNeo4j.hostname}:9999`)
+      expect(true).toBe('false')
```
Comment (robsdedude) [inline packages/neo4j-driver/test/driver.test.js]: What is this assertion?

File: packages/neo4j-driver/test/driver.test.js (hunk 5 above)
Comment (bigmontz) [reply packages/neo4j-driver/test/driver.test.js]: Good catch. I fixed the test to make it correct.

File: packages/neo4j-driver/test/driver.test.js (hunk 6 above)
Comment (bigmontz) [reply packages/neo4j-driver/test/driver.test.js]: Fixed

**Review by robsdedude — APPROVED** [review]
:soap:

**Pull request merged**